src/oemof/solph/_energy_system.py

```diff
@@ -192,6 +193,26 @@ def _extract_periods_years(self):
 
             self.periods_years = periods_years
 
+    def _extract_periods_matrix(self):
+        """Determines a matrix describing the temporal distance to each period.
+        Rows represent investment/commissioning periods, columns represent
+        decommissioning periods. The values describe the temporal distance
+        between each investment period to each decommissioning period.
+
+        Returns
+        -------
+        period_distance_matrix: np.array
+
+        """
+        periods_matrix = []
+        if self.periods is not None:
+            period_years = np.array(self.periods_years)
+            for k, v in enumerate(period_years):
+                row = period_years - v
+                row = np.where(row < 0, 0, row)
+                periods_matrix.append(row)
```

Comment:
Variable `k` is not used, but I see why you put it this way. It is elegant.

We might either change to `_` for assigning a value we don't need or use something like:
```python3
for i in range(period_years.shape[0]):
    row = period_years - period_years[i]
    row = np.where(row < 0, 0, row)
    periods_matrix.append(row)
```

Reply:
You are right, that's probably leftovers from changed code.